### ISSUE TITLE
Provider `v3` drops state params in Pact file with spec `v2`

### ISSUE THREAD

**Issue by martinslota** [pact-foundation/pact-js#1079]
### Software versions

- **OS**: Mac OS 13.2.1
- **Consumer Pact library**:  PactJS v 11.0.2
- **Provider Pact library**: N/A
- **Node Version**: v18.15.0

### Issue Checklist

Please confirm the following:

- [X] I have upgraded to the latest
- [X] I have the read the FAQs in the Readme
- [X] I have triple checked, that there are **no unhandled promises** in my code and have [read](https://github.com/pact-foundation/pact-js/blob/master/docs/troubleshooting.md#test-fails-when-it-should-pass) the section on intermittent test failures
- [X] I have set my log level to debug and attached a log file showing the complete request/response cycle
- [X] For bonus points and virtual high fives, I have created a reproduceable git repository (see below) to illustrate the problem

### Expected behaviour

When using the `v3` Pact provider with spec version 2, state parameters (as well as multiple states) should be present in the `providerState` field of interactions in the generated Pact file.

### Actual behaviour

Only the `description` of the first state is copied into the `providerState` field for each interaction.

### Steps to reproduce

See https://github.com/martinslota/pact-js-bug-state-params-dropped.

### Relevant files

* [test.log](https://github.com/pact-foundation/pact-js/files/11093685/test.log)
* [MyConsumer-MyProvider.json.txt](https://github.com/pact-foundation/pact-js/files/11093695/MyConsumer-MyProvider.json.txt)

**Comment by TimothyJones** [comment]
> When using the v3 Pact provider with spec version 2, state parameters (as well as multiple states) should be present in the providerState field of interactions in the generated Pact file.

I don't think this is the expected behaviour. Pact spec version 2 does not support state parameters.

**Comment by mefellows** [comment]
That's correct. Perhaps we shouldn't allow it in the DSL

**Comment by martinslota** [comment]
> I don't think this is the expected behaviour. Pact spec version 2 does not support state parameters or multiple states.

That's a good point.

The bug report originates from my effort to upgrade from `@pact-foundation/pact` version 9. The original code put an entire stringified JSON array into the `state` field of an interaction. I guess I expected that if I upgrade and put the original data into `states`, it bubbles all the way to the `providerState` field in the resulting pact file.

I suppose I could in principle put the stringified array into the `description` field of the first element of an interaction's `states` field and I'd get the effect I want.

As far as I'm concerned, we can close this bug report.

**Comment by martinslota** [comment]
> That's correct. Perhaps we shouldn't allow it in the DSL

That would be a very nice touch! 💯

**Comment by mefellows** [comment]
In the V4 interface, the API itself doesn't allow incompatible things to be done like this. It's a bit harder to do with the V3 API because we allow the spec as an option in the constructor and that changes what can/can't be done later in the API, making the checks complicated/messy.

I'll leave it open for now as a potential enhancement.
